src/app/wishlists/wishlists.ts

```diff
@@ -96,18 +106,19 @@ function getWishListPlugs(item: DimItem, wishListRoll: WishListRoll): Set<number
  * Disregards cosmetics and some other socket types.
  */
 function allDesiredPerksExist(item: DimItem, wishListRoll: WishListRoll): boolean {
-  if (!item.sockets) {
-    return false;
-  }
-
   if (wishListRoll.isExpertMode) {
-    for (const rp of wishListRoll.recommendedPerks) {
+    for (const recommendedPerk of wishListRoll.recommendedPerks) {
       let included = false;
 
-      outer: for (const s of item.sockets.allSockets) {
+      // this function serves only getInventoryWishListRoll,
+      // which has already ensured item.sockets exists
+      outer: for (const s of item.sockets!.allSockets) {
         if (s.plugOptions) {
-          for (const dp of s.plugOptions) {
-            if (dp.plugDef.hash === rp) {
+          for (const plug of s.plugOptions) {
+            if (
+              plug.plugDef.hash === recommendedPerk ||
+              perkToEnhanced[recommendedPerk] === plug.plugDef.hash
+            ) {
```

Comment:
I know this is pre-existing but these nested labelled loops could probably be more concise if replaced with `every`/`some`.

Reply:
i assumed this was done for performance or smth, i just kept the overarching structure and renamed it a bit. i think it's effectively a double, nested .some? i'd prefer to leave that to someone else to mess with.